### PR TITLE
Use Kurbo's Point and Affine types

### LINKED ISSUE #100: Use kurbo's Affine and Point types
Experimental adoption suggested a couple of issues:

* Affine has no into_inner *update* it's just not called that, https://docs.rs/kurbo/latest/kurbo/struct.Affine.html#method.as_coeffs
   * can't perform any operation kurbo didn't see fit to provide for you. Ex https://github.com/googlefonts/picosvg/blob/69cbfec486eca35a46187405abc39f608d3b2963/src/picosvg/svg_transform.py#L270.
   * Can't round for test stability
* No eq, hash
   * This is "correct" for float, we work around it with OrderedFloat today.
   * If we had into_inner we could likely get past this

### PR DESCRIPTION
**Pull request by rsheeter**
Minimize reinvention of wheel by using kurbo for transform and point.

Builds on #98. Fixes #100. Step toward #99.

### CODE REVIEW

File: glyphs-reader/src/font.rs [hunk 1]
```diff
@@ -410,7 +304,7 @@ impl TryFrom<BTreeMap<String, Plist>> for Component {
         let mut transform = if let Some(plist) = dict.remove("transform") {
             Affine::from_plist(plist)
         } else {
-            Affine::identity()
+            Default::default()
```
Comment (dfrg) [inline glyphs-reader/src/font.rs:307]: Purely personal preference, but as a reader, `Affine::IDENTITY` here would provide some clarity.

File: glyphs-reader/src/font.rs [hunk 2]
```diff
@@ -1206,6 +1100,43 @@ impl Font {
...
+struct AffineForEqAndHash([OrderedFloat<f64>; 6]);
+
+impl From<Affine> for AffineForEqAndHash {
+    fn from(value: Affine) -> Self {
+        let coeffs = value.as_coeffs();
+        let mut ordered_float_coeffs = [OrderedFloat::<f64>::from(0.0); 6];
+        for i in 0..6 {
+            ordered_float_coeffs[i] = coeffs[i].into();
+        }
+        AffineForEqAndHash(ordered_float_coeffs)
```
Comment (dfrg) [inline glyphs-reader/src/font.rs:1136]:
```suggestion
        Self(value.as_coeffs().map(|coeff| coeff.into()))
```

Thanks to const generics, [`map`](https://doc.rust-lang.org/std/primitive.array.html#method.map) now does wonderful things on arrays.

**Review by dfrg — APPROVED** [review]
Lgtm with a few style suggestions inline. Nice to see kurbo making a comeback!

note: the PR seemed to get updated as I was reviewing. I left those comments hoping they still might be useful.

File: glyphs2fontir/src/toir.rs [hunk 3]
```diff
@@ -1,13 +1,102 @@
...
+        }
+        let mut pt = to_ir_point(first);
+        pt.typ = ir::PointType::Move;
+        contour.push(pt);
+    } else {
+        // In Glyphs.app, the starting node of a closed contour is always
+        // stored at the end of the nodes list.
+        let Some((last, elements)) = nodes.split_last() else {
+            unreachable!("Not empty and no last is a good trick");
+        };
```
[the remark this answers is not in the document]
Comment (rsheeter) [reply glyphs2fontir/src/toir.rs]: I thought unreachable might be clearer but in seeing the expect spelled out I'm not sure that's true. Updated.

File: glyphs2fontir/src/toir.rs [hunk 4]
```diff
@@ -1,13 +1,102 @@
...
+    // See also https://github.com/fonttools/ufoLib2/blob/4d8a9600148b670b0840120658d9aab0b38a9465/src/ufoLib2/pointPens/glyphPointPen.py#L16
+    let mut contour = ir::Contour::new();
+    if path.nodes.is_empty() {
+        return Ok(contour);
+    }
+
+    let mut nodes = &path.nodes[..];
+    if !path.closed {
+        let first = &nodes[0];
+        nodes = &nodes[1..];
```
[the remark this answers is not in the document]
Comment (rsheeter) [reply glyphs2fontir/src/toir.rs]: Good catch, fixed.

File: glyphs-reader/src/font.rs (hunk 1 above)
Comment (rsheeter) [reply glyphs-reader/src/font.rs:307]: Ha, apparently I'm looking at a version of the kurbo docs prior that existing. Fixed.

File: glyphs2fontir/src/toir.rs [hunk 5]
```diff
@@ -1,13 +1,102 @@
...
+        // In Glyphs.app, the starting node of a closed contour is always
+        // stored at the end of the nodes list.
+        let Some((last, elements)) = nodes.split_last() else {
+            unreachable!("Not empty and no last is a good trick");
+        };
+        nodes = elements;
+        contour.push(to_ir_point(last));
+    }
+
+    nodes.iter().map(to_ir_point).for_each(|p| contour.push(p));
```
[the remark this answers is not in the document]
Comment (rsheeter) [reply glyphs2fontir/src/toir.rs]: Fixed, ty

**Comment by rsheeter** [comment]
> PR seemed to get updated as I was reviewing

Sorry, I goofed a branch slightly and naively assumed nobody else would work my silly hours so I could fix it and nobody would ever see :)

**Pull request merged**